app/controllers/events_controller.rb

```diff
@@ -2,97 +2,55 @@ class EventsController < ApplicationController
   before_filter :authenticate_user!
   load_and_authorize_resource
 
-  def schedule
-    @event = Event.find(params[:id])
-    @person = Person.find(params[:person_id])
-    if @person and @event.ready_to_schedule?(params[:card_action])
-      @tc = @event.schedule(@person, params[:card_action])
-    end
-
-    redirect_to event_url(@event), status: :found, notice: @tc ? "Timecard created" : "Timecard not created"
-  end
-
-  # GET /events
-  # GET /events.json
   def index
     # by default show all scheduled or in-session events
-    @events = params['all_events'] == "true" ? Event.all : Event.where('end_time > ?', Time.now)
+    @events = params['all_events'] == "true" ? Event.all : Event.actual.where('end_time > ?', Time.now)
     @page_title = params['all_events'] == "true" ? "All Events" : "Current Events"
-    respond_to do |format|
-      format.html # index.html.erb
-      format.json { render json: @events }
-    end
   end
 
-  # GET /events/1
-  # GET /events/1.json
   def show
-    @event = Event.find(params[:id])
```

Comment:
No, because in line 2 we have a before filter for the authorization. Since the authorization is based on the object the user is trying to access, this before filter loads the object for us.

An example is here
https://github.com/ReadyResponder/ReadyResponder/blob/development/app/controllers/items_controller.rb#L10